src/main/java/com/google/sps/tool/WordSearch.java

```diff
@@ -0,0 +1,104 @@
+package com.google.sps.tool;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.Collections;
+import java.util.Random;
+
+public class WordSearch {
+  private static HashMap<Character, ArrayList<String>> alphabetWordMap =
+      new HashMap<Character, ArrayList<String>>();
+
+  public static ArrayList<String> generateWordOptions(String correctAnswer) {
+    ArrayList<String> quizWords = new ArrayList<>();
+    quizWords.add(correctAnswer);
+
+    if (!alphabetWordMap.containsKey('a') && !initMap()) {
+      // If the dictionary file can't be read then it just return the correct Answer
+        return quizWords;
+    }
+
+    char firstLetter = correctAnswer.charAt(0);
+    ArrayList<String> wordList = alphabetWordMap.get(Character.toLowerCase(firstLetter));
+    Boolean wordFound = false;
+
+    for (int i = 0; i < wordList.size(); i++) {
+      if (wordList.get(i).equals(correctAnswer)) {
+        wordFound = true;
+        getRandomWords(i, wordList, quizWords);
+        break;
+      }
+    }
+
+    // If exact word could not be found, then generate random words with the same first letter
+    if (!wordFound) {
+      Random random = new Random();
+      for (int i = 0; i < 3; i++) {
+        int index = random.nextInt(wordList.size());
+        quizWords.add(wordList.get(index));
+      }
+    }
+
+    Collections.shuffle(quizWords);
+    return quizWords;
+  }
+
+  private static void getRandomWords(
+      Integer origin, ArrayList<String> wordList, ArrayList<String> quizWords) {
+    Random random = new Random();
+    for (int i = 0; i < 3; i++) {
+      try {
+        int index = random.nextInt(20) - 10;
```

Comment:
Clamp this index between 0 and `wordList.length - 1` instead of catching the exception.

https://twitter.com/jaffathecake/status/1296382880030044160

Reply:
done.

Reply:
done.